futures-util/src/stream/stream/flatten_unordered.rs

```diff
@@ -424,15 +400,24 @@ where
 
                     break;
                 } else {
-                    match this.stream.as_mut().poll_next(&mut Context::from_waker(&stream_waker)) {
+                    // Initialize base stream waker if it's not yet initialized
+                    if stream_waker.is_none() {
+                        // Safety: now state is `POLLING`.
+                        stream_waker
+                            .replace(unsafe { WrappedWaker::replace_waker(this.stream_waker, cx) });
+                    }
+                    let mut cx = Context::from_waker(stream_waker.as_ref().unwrap());
+
+                    match this.stream.as_mut().poll_next(&mut cx) {
                         Poll::Ready(Some(inner_stream)) => {
+                            let next_item_fut = PollStreamFut::new(Box::pin(inner_stream));
```

Comment:
Hmm. I feel removing Unpin bound by boxing each stream is worse than requiring Unpin bound; if the stream is already Unpin, it adds an extra cost.

Reply:
I'd say this combinator is much more convenient without `Unpin` requirement, but let's choose the pragmatic performance over the usability)